ambry-frontend/src/test/java/com.github.ambry.frontend/AmbrySecurityServiceTest.java

```diff
@@ -299,20 +302,63 @@ private void verifyAbsenceOfHeaders(MockRestResponseChannel restResponseChannel,
 
   /**
    * Tests {@link SecurityService#processResponse(RestRequest, RestResponseChannel, BlobInfo, Callback)} for a Get blob
-   * with the passed in {@link BlobInfo}
+   * with the passed in {@link BlobInfo} and various range settings, including no set range (entire blob).
+   * @param blobInfo the {@link BlobInfo} to be used for the {@link RestRequest}s
+   * @throws Exception
+   */
+  private void testGetBlobWithVariousRanges(BlobInfo blobInfo)
+      throws Exception {
+    long blobSize = blobInfo.getBlobProperties().getBlobSize();
+    testGetBlob(blobInfo, null);
+
+    testGetBlob(blobInfo, ByteRange.fromLastNBytes(0));
+    if (blobSize > 0) {
+      testGetBlob(blobInfo, ByteRange.fromStartOffset(0));
+      testGetBlob(blobInfo, ByteRange.fromStartOffset(ThreadLocalRandom.current().nextLong(1, blobSize - 1)));
+      testGetBlob(blobInfo, ByteRange.fromStartOffset(blobSize - 1));
+
+      long random1 = ThreadLocalRandom.current().nextLong(blobSize);
+      long random2 = ThreadLocalRandom.current().nextLong(blobSize);
+      testGetBlob(blobInfo, ByteRange.fromOffsetRange(Math.min(random1, random2), Math.max(random1, random2)));
+
+      testGetBlob(blobInfo, ByteRange.fromLastNBytes(ThreadLocalRandom.current().nextLong(blobSize)));
```

Comment:
don't you want to make sure this won't be 0 (like you have done with start offset)? Otherwise you might end up repeating a test you have already done.